pymeasure/display/curves.py

```diff
@@ -23,15 +23,23 @@
 #
 
 import logging
+import sys
 
 import pyqtgraph as pg
 import numpy as np
+try:
+    from matplotlib.cm import viridis
+except ImportError:
+    log.warning("Matplotlib not found. Images will be greyscale")
```

Comment:
The `log` variable is called before construction. That section (currently lines 37-38) needs to be above this one.